service/src/test/java/bio/terra/workspace/service/resource/ValidationUtilsTest.java

```diff
@@ -239,4 +239,54 @@ public void validateBqDataTableName_invalidName_throwsException() {
         InvalidNameException.class, () -> ValidationUtils.validateBqDataTableName("00_お客様*"));
     assertThrows(InvalidNameException.class, () -> ValidationUtils.validateBqDataTableName(""));
   }
+
+  @Test
+  public void validateGitHubRepoHttpsUrl() {
+    ValidationUtils.validateGitRepoUrl(
+        "https://github.com/DataBiosphere/terra-workspace-manager.git");
+    ValidationUtils.validateGitRepoUrl("https://github.com/yuhuyoyo/testrepo.git");
+    ValidationUtils.validateGitRepoUrl("git@github.com:DataBiosphere/terra-workspace-manager.git");
+    ValidationUtils.validateGitRepoUrl("git@github.com:yuhuyoyo/testrepo.git");
+    ValidationUtils.validateGitRepoUrl(
+        "https://username:password@github.com/username/repository.git");
+  }
+
+  @Test
+  public void validateGitHubRepoHttpsUrl_httpUrl_throwsException() {
+    assertThrows(
+        InvalidReferenceException.class,
+        () ->
+            ValidationUtils.validateGitRepoUrl(
```

Comment:
It's not obvious why some of these are invalid. Is it because they're missing a .git?

Reply:
this one is because it's a http url instead of https. the one below is missing .git in the end. I will update the test name to make it more obvious.